File: netty/py_thrift/PersonServiceImpl.py
# -*- coding:utf-8 -*-
__author__ = '作者'
from py.thrift.generated import PersonService
from py.thrift.generated import ttypes
from thrift import Thrift
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TCompactProtocol

class PersonServiceImpl:
    def getPersonByUsername(self,username):
        print("got client parma: " + username)

        person = ttypes.Person()
        person.username = username
        person.age = 18
        person.married = False
        return person
    def savePerson(self,person):
        print("got client parma: ")
        print(person.username)
        print(person.age)
        print(person.married)

File: netty/netty_demo/src/main/java/hhz/demo08/PersonServiceImpl.java
package hhz.demo08;

import hhz.demo08.thrift.generated.DataException;
import hhz.demo08.thrift.generated.Person;
import hhz.demo08.thrift.generated.PersonService;
import org.apache.thrift.TException;

/**
 * @ClassName: PersonServiceImpl
 * @Description: TODO(概况)
 * @author: huanghz
 * @date: 2019/9/3 下午 01:04
 */
public class PersonServiceImpl implements PersonService.Iface
{
    @Override
    public Person getPersonByUsername(String username) throws DataException, TException
    {
        System.out.println("Got Client Param: " + username);
        Person person = new Person();
        person.setUsername(username);
        person.setAge(20);
        person.setMarried(false);
        return person;
    }

    @Override
    public void savePerson(Person person) throws DataException, TException
    {
        System.out.println(person.getUsername());
        System.out.println(person.getAge());
        System.out.println(person.isMarried());
    }
}
